test/powershell/Modules/Microsoft.PowerShell.Management/Set-Location.Tests.ps1

```diff
@@ -231,3 +231,58 @@ Describe "Set-Location" -Tags "CI" {
         }
     }
 }
+
+Describe "Set-Location: Name with special/wildcards characters" -Tags "CI" {
+
+    BeforeAll {
+
+        $startLocation = (Get-Location).Path
+
+        # printable ascii characters excluding alphanumerics
+        $specialChars =  ' !"#$%&''()*+,-./:;<=>?@[\]^_`{|}~	'.ToCharArray()
+        $testCases = @()
+        ForEach ($specialChar in $specialChars) {
+            If ($specialChar -ne '.') { $testCases += @{Name=([string] $specialChar)} }
+            $testCases += @{Name=('foo' + $specialChar)}
+            $testCases += @{Name=($specialChar + 'foo')}
+        }
+    }
+
+    It "Should be able to create and set location to <Name>" -TestCases $testCases {
+        param ( $Name )
+
+        # skip unsupported characters for the current platform 
+        if ($Name.IndexOfAny([System.IO.Path]::GetInvalidFileNameChars()) -ge 0 -or `
+            ($IsWindows -and ($Name.EndsWith('.') -or $Name.EndsWith(' ')))) {
+            Set-ItResult -Skipped -Because "'${Name}' is not supported as directory name on this operating system."
+            return
+        }
+
+        # note processing of '\' should be allowed on posix systems but 
+        # is quite broken in many powershell cmdlets
+        if ($Name.Contains('\')) 
+        {
+            Set-ItResult -Skipped -Because "path elements with backslashes are not fully supported in PowerShell on this operating system."
+            return
+        }
+
+        $dirFullName = (Join-Path $TestDrive $Name) 
+        
+        $newDir = [System.IO.Directory]::CreateDirectory($dirFullName)
+        $newDir.Name | Should -BeExactly $Name
+        
+        $newDirGetCheck = Get-Item -LiteralPath $dirFullName -Force
+        $newDirGetCheck.Name | Should -BeExactly $Name
+
+        try {
+            Set-Location -LiteralPath $dirFullName
+            [System.IO.Path]::GetFileName((Get-Location).Path) | Should -BeExactly $Name
```

Comment:
Please make sure there is an issue that this cannot be done with Split-Path.

Reply:
Already issue filed: https://github.com/PowerShell/PowerShell/issues/8751